### PR TITLE
Implement osdp_MFG and osdp_MFGREP

### PR DESCRIPTION
**Pull request by bjyoungblood**

### CODE REVIEW

File: lib/jeff/mfg_command.ex [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  The `Jeff.MfgCommand` protocol converts an Elixir data structure into an
+  osdp_MFG command.
+
+  ### Example
+
+      defmodule InputDisable do
+        defstruct input_number: nil, duration: nil
+
+        defimpl Jeff.MfgCommand do
+          def vendor_code(_), do: 0xC0FFEE
```
Comment (jjcarstens) [inline lib/jeff/mfg_command.ex:12]: ☕

File: lib/jeff/mfg_command.ex [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  @doc """
+  Returns the 3-byte vendor code associated with the command.
+  """
+  @spec vendor_code(t()) :: Jeff.vendor_code()
+  def vendor_code(command)
+
+  @doc """
+  Encodes the given `command` as a binary.
+  """
+  @spec encode(t) :: Jeff.Command.Mfg.t()
```
Comment (jjcarstens) [inline lib/jeff/mfg_command.ex:32]:
```suggestion
  @spec encode(t) :: binary()
```

Should this be speced to return binary?

**Review by jjcarstens — COMMENTED** [review]
Using a protocol could actually be a good use here. Thought I don't love `Jeff.MfgCommand` as the protocol name when there is also `Jeff.Command.Mfg` ...but I don't have any better ideas right now 🤔

**Comment by bjyoungblood** [comment]
> Using a protocol could actually be a good use here. Thought I don't love `Jeff.MfgCommand` as the protocol name when there is also `Jeff.Command.Mfg` ...but I don't have any better ideas right now 🤔

@jjcarstens thoughts on `Jeff.MfgCommandEncoder` or `Jeff.Command.Mfg.Encoder`?

**Comment by jjcarstens** [comment]
How about `Jeff.MFG.Encoder`?

**Review by jjcarstens — APPROVED** [review]

**Pull request merged**